### PR TITLE
build/ocp: sanitize and add in some status

### PR DESCRIPTION
**Pull request by sosiouxme**
Removed some cruft, changed some things to seem more sane (like using
params.VAR instead of manually converting env.VAR).

Also added updates to job information to make it easier to see at a
glance what happened with a particular build in the list of builds.

### CODE REVIEW

**Comment by sosiouxme** [comment]
@adammhaile @tbielawa

File: jobs/build/ocp/Jenkinsfile [hunk 1]
```diff
@@ -137,12 +151,20 @@ online:int                {origin,origin-web-console,openshift-ansible}/master -
     ]
 )
 
-IS_TEST_MODE = TEST.toBoolean()
+// User can have the job end if this is just a run to pick up parameter changes
+// (which Jenkins discovers by running the job).
+if (env.MOCK == null || params.MOCK) {
+    currentBuild.displayName = "#${currentBuild.number} - update parameters"
+    currentBuild.description = "Ran in mock mode"
+    return
```
Comment (tbielawa) [inline jobs/build/ocp/Jenkinsfile]: NICE

File: pipeline-scripts/commonlib.groovy [hunk 2]
```diff
@@ -16,6 +16,16 @@ def initialize() {
 
 }
 
+def safeArchiveArtifacts(List patterns) {
```
Comment (tbielawa) [inline pipeline-scripts/commonlib.groovy]: NICE

**Pull request merged**